drv/transceivers-server/src/main.rs

```diff
@@ -127,7 +129,7 @@ struct ServerImpl {
     consecutive_nacks: [u8; NUM_PORTS as usize],
 
     /// Handle to write thermal models and presence to the `thermal` task
-    thermal_api: Thermal,
+    thermal_api: Option<Thermal>,
```

Comment:
this means that we're checking at runtime whether thermal control is enabled, but whether or not it's enabled is determined at compile time by the feature flag. so, we could instead write
```suggestion
    #[cfg(feature = "thermal-control")]
    thermal_api: Thermal,
```
to avoid the runtime checks